### PR TITLE
update to k2pdfopt 2.55

### PR DESCRIPTION
**Pull request by benoit-pierre**
Note: I disabled the builtin support for fetching tesseract language data directly from Github: it's insecure (`--no-check-certificate`), relies on calling `wget` or `curl` directly, and not really useful without a hook so koreader' side can make sure Wi-Fi is enabled.

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/libk2pdfopt/49)

### CODE REVIEW

File: k2pdfoptlib/bmpregion.c [hunk 1]
```diff
@@ -640,6 +683,47 @@ exit(10);
     */
     }
 
+/*
+** v2.52
+** For each column, only count the top-most and bottom-most black pixels in that column
+** into the rowcount[] stats.  This prevents lowercase height from getting fooled by
+** letters like e, a, s, A, H.
+**
+** DECIDED THIS DIDN'T REALLY HELP
```
Comment (Frenzie) [inline k2pdfoptlib/bmpregion.c:692]: Out of curiosity, who decided this?

Comment (benoit-pierre) [reply k2pdfoptlib/bmpregion.c:692]: Let me check… (TODO: update the `update_upstream.sh*` script to use a dedicated `k2pdfopt` commit identity).

**Review by Frenzie — APPROVED** [review]
It's a bit hard to read the most important diff but the changes sound fine.

File: k2pdfoptlib/bmpregion.c (hunk 1 above)
Comment (benoit-pierre) [reply k2pdfoptlib/bmpregion.c:692]: This come from upstream.

Comment (Frenzie) [reply k2pdfoptlib/bmpregion.c:692]: Right, thanks.

**Comment by benoit-pierre** [comment]
Yeah, here's the diff with upstream:
- [before](https://github.com/koreader/libk2pdfopt/compare/9965188256407d2ab272dc8bee21aab538139869..563b93b61947849783355fe201bab47b9e9ff4d6)
- [after](https://github.com/koreader/libk2pdfopt/compare/upstream..benoit-pierre:libk2pdfopt:pr/update_to_k2pdfopt_2.55)

File: k2pdfoptlib/bmpregion.c (hunk 1 above)
Comment (benoit-pierre) [reply k2pdfoptlib/bmpregion.c:692]: From 2.52 to be precise.

**Pull request merged**